File: backend/secureattend-backend/src/main/java/com/secureattend/factory/CodeFactory.java
package com.secureattend.factory;

import org.springframework.stereotype.Component;
import java.security.SecureRandom;
import java.util.UUID;

@Component
public class CodeFactory {
    private static final String ALPHANUM = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
    private static final SecureRandom RANDOM = new SecureRandom();

    public String generateQrToken() {
        return "QR_" + UUID.randomUUID().toString().substring(0, 12).toUpperCase();
    }

    public String generateCodeword() {
        StringBuilder sb = new StringBuilder(6);
        for (int i = 0; i < 6; i++) {
            sb.append(ALPHANUM.charAt(RANDOM.nextInt(ALPHANUM.length())));
        }
        return sb.toString();
    }
}

File: backend/secureattend-node/src/utils/codeFactory.js
import { randomUUID } from 'crypto';

const ALPHANUM = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';

export function generateQrToken() {
  return 'QR_' + randomUUID().replace(/-/g, '').substring(0, 12).toUpperCase();
}

export function generateCodeword() {
  let word = '';
  for (let i = 0; i < 6; i++) {
    word += ALPHANUM.charAt(Math.floor(Math.random() * ALPHANUM.length));
  }
  return word;
}
